jython-processors/trigonometric-transformation/processor.py

```diff
@@ -1,20 +1,117 @@
+        return math.pi/2.0-(math.pi*value)/(self.period/2.0)
+    
+    def compute_trigonometric_features(self):
+        for value in self.range_period:
+            value_argument = self.compute_argument(value)
+            self.trigonometric_features['cos'][value] = self.trigonometric_transformer['cos'](value_argument)
+            self.trigonometric_features['sin'][value] = self.trigonometric_transformer['sin'](value_argument)
+            self.trigonometric_features['tan'][value] = self.trigonometric_transformer['tan'](value_argument)
+            pass
+            
+#Parameters import :
+periodical_column_name = params.get('periodical_column')
+process_as_time_dimension = params.get('process_as_time_dimension')
+column_period = params.get('column_period')
+range_period = range(int(column_period+2))
+compute_cos = params.get('compute_cos')
+compute_sin = params.get('compute_sin')
+compute_tan = params.get('compute_tan')
+output_arg = params.get('output_arg')
+L_choosed_functions = [compute_cos, compute_sin, compute_tan]
+L_functions = ['cos', 'sin', 'tan']
+L_trigonometric_cols = []
+
+if output_arg:
+    L_trigonometric_cols.append(periodical_column_name + "_arg")
+    
+for choosed_function, function in zip(L_choosed_functions, L_functions):
+    if choosed_function :
+        L_trigonometric_cols.append(periodical_column_name+"_"+function)
+
+               
+trigonometric_transformer = TrigonometricTransformer(L_choosed_functions=L_choosed_functions,
+                                                     period=column_period,
+                                                     process_as_time_dimension=process_as_time_dimension,
+                                                     range_period=range_period)
+
+trigonometric_features = trigonometric_transformer.trigonometric_features
+
+#Production code :
+def process(row):
+    periodical_column_value = row[periodical_column_name]    
+    try:
+        periodical_column_value = int(periodical_column_value)
+        arg_val = trigonometric_transformer.compute_argument(periodical_column_value)
+        if process_as_time_dimension:
+            cos_val = trigonometric_features['cos'][periodical_column_value]
+            sin_val = trigonometric_features['sin'][periodical_column_value]
+            tan_val = trigonometric_features['tan'][periodical_column_value]
+        else:
+            cos_val, sin_val, tan_val = trigonometric_transformer.compute_trigonometric_transform(arg_val)
+        log_message = None
+        
+    except Exception as e:
+        log_message = "ERROR : "+str(e)
+        arg_val, cos_val, sin_val, tan_val  = None, None, None, None
     
-    cos_name = periodical_column_name + "_cos"
-    sin_name = periodical_column_name + "_sin"
-    tan_name = periodical_column_name + "_tan"
-    cos_val = math.cos(arg_val)
-    sin_val = math.sin(arg_val)
-    tan_val = math.tan(arg_val)
+    L_values = [arg_val, cos_val, sin_val, tan_val]
```

Comment:
use valid english variables names following this_casing_example.
```suggestion
    output_values = [arg_val, cos_val, sin_val, tan_val]
```

Reply:
Edit done : variables updated everywhere in the code.

Reply:
Edit : The code structure changed : we do not use anymore this variable